truetype/truetype.go

```diff
@@ -201,13 +201,18 @@ func (f *Font) parseCmap() error {
 		return err
 	}
 	offset = int(u32(f.cmap, offset+4))
-	if offset <= 0 || offset > len(f.cmap) {
+	cmapLength := len(f.cmap)
+	if offset <= 0 || offset+2 > cmapLength {
 		return FormatError("bad cmap offset")
 	}
 
 	cmapFormat := u16(f.cmap, offset)
 	switch cmapFormat {
 	case cmapFormat4:
+		// check cmap length for reading 2+2 bytes starting at offset+4
```

Comment:
I'm not sure if these comments add much. I'd just say

if offset > len(f.cmap) - 8 {